Reg-ex based email address validation

er1c opened the pull request:
Generic utility method useful for form validation of an "Email" type.

tpunder commented on src/main/scala/fm/common/EmailSender.scala:
```diff
@@ -6,6 +6,17 @@ import javax.mail.{Message, Session, Transport}
+object EmailSender {
+  // RegEx From: https://www.w3.org/TR/html5/forms.html#valid-e-mail-address
+  //   Modified to require at least one .<tld> at the end, so root@localhost isn't valid (changed *$""") => +$""")
+  private val emailRegex = """^[a-zA-Z0-9\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+$""".r
```
Add type to this

tpunder commented on src/test/scala/fm/common/TestEmailSender.scala:
```diff
@@ -0,0 +1,17 @@
+    isValidEmail("t es t@gmailcom")      should equal(false)
+    isValidEmail("b ob @tes tmai l.com") should equal(false)
+    isValidEmail("test@gmailcom")        should equal(false)
+  }
```
It would probably be nice to have some more complex tests cases

er1c commented:
PR Updated, added the Regex type/import and added some more complicated test cases, still technically have a few that would fail this approach, but probably fine for now
